support/builder/build_dep_populate.sh

```diff
@@ -0,0 +1,463 @@
+                    echo "Setting endpoint to ${minioIP//\"}:9000"
+                    if curl -s localhost:9631/services/builder-minio/"${i}" | jq .cfg.use_ssl | grep "true"> /dev/null; then
+                        export AWS_REGION="https://${minioIP//\"}:9000"
+                    else
+                        export AWS_REGION="http://${minioIP//\"}:9000"
+                    fi
+                    return
+                else
+                    echo ""
+                    echo "==========================================================="
+                    echo "Please enter the minio endpoint URI and press [ENTER]:"
+                    echo "(http://localhost:9000 || https://10.1.250.4:9000)"
+                    echo "==========================================================="
+                    read -r region_name
+                    export AWS_REGION=${region_name}
+                    return
+                fi
+            fi
+        done
+    else
+        echo ""
+        echo "==========================================================="
+        echo "Please enter the region for your bucket and press [ENTER]:"
+        echo "(us-west-1 us-east-1 us-west-2 etc )"
+        echo "==========================================================="
+        read -r region_name
+        export AWS_REGION=${region_name}
+    fi
+}
+
+failed_uploads=()
+uploadHarts() {
+    # Takes the artifact array generated via
+    # traversing the data path and uploads to s3
+    uploadMsg
+    putArtifacts
+    echo ""
+    echo "########################################"
+    echo "${#artifacts[@]} Artifacts have been ingested!"
+    echo "########################################"
+    echo ""
+
+    if (( "${failed_uploads[@]}" )); then
+    echo ""
+    echo "########################################"
+    echo "The following artifacts failed on upload:"
+    for failure in "${failed_uploads[@]}"; do
+        echo "${failure}"
+    done
+    echo "########################################"
+    fi
+}
+
+putArtifacts() {
+    # shellcheck disable=SC2044
+    for file in $(find /hab/tmp/harts/ -name '*.hart'); do
+        if ! hab pkg upload --url "${bldr_url}" "${file}" --force; then
+            failed_uploads+=("${bldr_url}")
+        fi
+    done
+}
+
+genS3Config() {
+    echo ""
+    read -r -p "AWS Access Key ID: " access_key_id
+        export AWS_ACCESS_KEY_ID="${access_key_id}"
+    read -r -p "AWS Secret Access Key: " secret_access_key
+        export AWS_SECRET_ACCESS_KEY="${secret_access_key}"
+}
+
+genBldrCreds() {
+    echo ""
+    echo "Please provide a Builder Auth Token."
+    read -r -p "Builder Auth token: " bldr_auth_token
+        export HAB_AUTH_TOKEN="${bldr_auth_token}"
+    echo ""
+    echo "Builder auth token set to: ${HAB_AUTH_TOKEN}"
+    read -r -p "Would you like to use this auth token? [y/N]" response
+    if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+        return
+    else
+        genBldrCreds
```

Comment:
Similar to above, this seems like a loop which I don't understand the purpose of

Reply:
see above